src/dlls/mscoree/unixinterface.cpp

```diff
@@ -146,6 +138,61 @@ int coreclr_initialize(
     ConstWStringArrayHolder propertyValuesW;
     propertyValuesW.Set(StringArrayToUnicode(propertyCount, propertyValues), propertyCount);
 
+    // Process properties that specify CoreCLR startup flags and remove them from the property list
+    STARTUP_FLAGS startupFlags =
+        static_cast<STARTUP_FLAGS>(
+            STARTUP_FLAGS::STARTUP_LOADER_OPTIMIZATION_SINGLE_DOMAIN |
+            STARTUP_FLAGS::STARTUP_SINGLE_APPDOMAIN);
+    int newPropertyIndex = 0;
+    for (int oldPropertyIndex = 0; oldPropertyIndex < propertyCount; ++oldPropertyIndex)
+    {
+        if (wcscmp(propertyKeysW[oldPropertyIndex], W("CONCURRENT_GC")) == 0)
+        {
+            if (wcscmp(propertyValuesW[oldPropertyIndex], W("1")) == 0)
+            {
+                startupFlags = static_cast<STARTUP_FLAGS>(startupFlags | STARTUP_CONCURRENT_GC);
```

Comment:
I wonder if that wouldn't be confusing, since no other options are on by default. And we never have tried to use the concurrent GC before, so it would require thorough testing before we would enable it by default.